src/main/java/com/ldtteam/blockout/views/TabSet.java

```diff
@@ -0,0 +1,271 @@
+     * i.e. the button you click to change content.
+     * Has a background and a potential icon or text otherwise
+     */
+    public class ButtonTab extends View
+    {
+        protected Tab   toSelect;
+        protected Image background = new Image();
+        protected Image icon = new Image();
+
+        public ButtonTab(Tab tab)
+        {
+            super();
+            int index = tabs.children.size();
+
+            tabs.addChild(this);
+            toSelect = tab;
+
+            setSize(tabWidth, tabHeight);
+            setPosition(
+              vertical ? 0 : index*(tabWidth + spacing),
+              vertical ? index*(tabHeight + spacing) : 0
+            );
+
+            if (tabIcons.length > 0)
+            {
+                background.setSize(width, height);
+                background.setImage(tabIcons[RANDOM.nextInt(tabIcons.length)], 0, 0, this.width, this.height);
+                addChild(background);
+            }
+
+            if (tab.icon != null)
+            {
+                Tuple<Integer, Integer> dim = Image.getImageDimensions(tab.icon);
+                if (width < dim.getA())
+                {
+                    int scale = width / dim.getA();
+                    icon.setSize(dim.getA() * scale, dim.getB() * scale);
+                }
+                if (height < icon.getHeight())
+                {
+                    int scale = height / icon.getHeight();
+                    icon.setSize(icon.getWidth() * scale, icon.getHeight() * scale);
+                }
+                else
+                {
+                    icon.setSize(dim.getA(), dim.getB());
+                }
+
+                icon.setAlignment(Alignment.MIDDLE);
+                icon.setPosition(vertical ? -overlap/4 : 0, vertical ? 0 : -overlap/4);
+                icon.setImage(tab.icon);
+                addChild(icon);
+
+                tab.createTooltip(this);
+            }
+            else
+            {
+                Text text = tab.buildLabel();
+                text.setAlignment(Alignment.MIDDLE);
+                text.setTextAlignment(Alignment.MIDDLE);
+                if (vertical)
+                {
+                    text.setPosition(-overlap/4, 0);
+                    text.setSize(width - overlap/2, height);
+                }
+                else
+                {
+                    text.setPosition(0, -overlap/4);
+                    text.setSize(width, height-overlap/2);
+                }
+                addChild(text);
+            }
+        }
+
+        @Override
+        public void drawSelf(final MatrixStack ms, final double mx, final double my)
+        {
+            if (toSelect.equals(selectedTab)) return;
+            super.drawSelf(ms, mx, my);
+        }
+
+        @Override
+        public void drawSelfLast(final MatrixStack ms, final double mx, final double my)
+        {
+            if (toSelect.equals(selectedTab)) super.drawSelf(ms, mx, my);
```

Comment:
This is a no, this pass should used for overlays which needs to sit on top of screen, not for delayed rendering. Use small z translate in drawSelf instead, if you wanted to achieve this being higher than base element - I would like to know why this even exists